### DISCUSSION TITLE
TurnTraceSwitchOff missing in DataConnection

### DISCUSSION THREAD

**Discussion by VarunSaiTeja** [linq2db/linq2db#3095] [Q&A] [answered]
I want to debug the trace occasionally.
So I want to switch on/off the trace based on the scenario.
I don't want to keep the switch on all time, Because I do a lot of bulk inserts which takes time to generate traces. So I want to turn off trace in specific scenarios.

What I found is DataConnection class only had TurnTraceSwitchOn method.
So is there any way to turn off the trace?

### Environment details
linq2db version: 3.4.0
Database Server: Sqlite
Database Provider: Microsoft.Data.Sqlite
Operating system: Windows 10
.NET version: 5

**Comment by jods4** [comment, marked as answer]
`TurnTraceSwitchOn` has an optional `TraceLevel` parameter you can pass.

To turn it off, this should work:
```csharp
DataConnection.TurnTraceOn(TraceLevel.Off);
```

I admit that the "Turn trace _on_" naming threw me off at first as well.
